### PR TITLE
Test rds instance deletion protection

### PR DESCRIPTION
**Pull request by Khushi-Kalantri-Comprinno**
### Context
This change adds unit tests for the rds_instance_deletion_protection check, which validates whether Amazon RDS instances have deletion protection enabled.

### Description
The test suite includes three cases:
With Deletion Protection Enabled: Verifies the check returns PASSED when deletion protection is set to True.
Without Deletion Protection: Verifies the check returns FAILED when deletion protection is False.
No RDS Instances Present: Verifies the check returns NOT_APPLICABLE when no RDS instances are found in the account.

### Checklist
- Added new check tests and reviewed necessary permissions (read-only access to rds:DescribeDBInstances)
-Code covered by unit tests (positive, negative, and edge cases)
-Documentation follows [Google Python Style Guide](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings)
-Considered if backporting is needed — Not required

### License
I confirm that my contribution is made under the terms of the **Apache 2.0 license**.

### CODE REVIEW

File: library/aws/tests/vpc/test_vpc_flowlogs_enable_logging.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        self.mock_client.describe_flow_logs.side_effect = [
+            {"FlowLogs": [{"FlowLogId": "fl-abc"}]},
+            {"FlowLogs": []}
+        ]
+
+        report = self.check.execute(self.mock_session)
+
+        assert report.status == CheckStatus.FAILED
+        assert any(r.status == CheckStatus.PASSED for r in report.resource_ids_status)
+        assert any(r.status == CheckStatus.FAILED for r in report.resource_ids_status)
```
Comment (aafaq-rashid-comprinno) [inline library/aws/tests/vpc/test_vpc_flowlogs_enable_logging.py:72]: <h3>🔧 <strong>Areas for Improvement</strong></h3>
<h4>1. ⚠️ <strong>Missing Exception/Error Handling Test</strong></h4>
<p>Unlike the <code inline="">iam_password_policy_lowercase</code> test, this one <strong>does not test error handling scenarios</strong>, such as:</p>
<ul>
<li>
<p><code inline="">describe_vpcs()</code> or <code inline="">describe_flow_logs()</code> raises <code inline="">ClientError</code></p>
</li>
<li>
<p><code inline="">describe_flow_logs()</code> raises a generic <code inline="">Exception</code></p>
</li>
</ul>
<p>✅ <strong>Recommendation</strong>: Add this test:</p>
<pre><code class="language-python">from botocore.exceptions import ClientError

def test_vpc_describe_flow_logs_client_error(self):
    """Test when describe_flow_logs raises a ClientError."""
    self.mock_client.describe_vpcs.return_value = {
        "Vpcs": [{"VpcId": "vpc-xyz"}]
    }
    self.mock_client.describe_flow_logs.side_effect = ClientError(
        error_response={"Error": {"Code": "UnauthorizedOperation", "Message": "Access denied"}},
        operation_name="DescribeFlowLogs"
    )

    report = self.check.execute(self.mock_session)

    assert report.status == CheckStatus.UNKNOWN
    assert report.resource_ids_status[0].status == CheckStatus.UNKNOWN
    assert "Error checking Flow Logs" in report.resource_ids_status[0].summary
</code></pre>
<p>Also add a test for a failure in <code inline="">describe_vpcs()</code>.</p>
<hr>
<h4>2. ✅ <strong>Mocking Setup and Code Readability</strong></h4>
<p>While functional, the setup is slightly verbose and could be more concise.</p>
<p>✅ <strong>Recommendation</strong>: Simplify and improve readability:</p>
<pre><code class="language-python">def setup_method(self):
    self.check = vpc_flowlogs_enable_logging(metadata={})
    self.mock_session = MagicMock()
    self.mock_ec2 = MagicMock()
    self.mock_sts = MagicMock()

    def client_selector(service_name):
        if service_name == "ec2":
            return self.mock_ec2
        elif service_name == "sts":
            return self.mock_sts
        raise Exception("Unknown service")

    self.mock_session.client.side_effect = client_selector
    self.mock_sts.get_caller_identity.return_value = {"Account": "123456789012"}
</code></pre>
<p>Also: consistent naming like <code inline="">mock_ec2</code> instead of <code inline="">mock_client</code> improves clarity.</p>
<hr>
<h4>3. 🧪 <strong>Assertions: Summary and Resource Count</strong></h4>
<p>Assertions verify <code inline="">status</code> and <code inline="">count</code>, but they <strong>don't verify <code inline="">summary</code> text content</strong> like in your IAM password policy tests.</p>
<p>✅ <strong>Recommendation</strong>: Add <code inline="">summary</code> checks:</p>
<pre><code class="language-python">assert "Flow Logs enabled" in report.resource_ids_status[0].summary
</code></pre>
<p>Or even loop through them to verify expected summaries per VPC.</p>
<hr>
<h4>4. ⚠️ <strong>No Use of <code inline="">pytest.mark.parametrize</code></strong></h4>
<p>This isn't strictly required, but for repeated tests with only input/output variations, parametrize can reduce duplication and improve readability.</p>
<p>✅ Optional idea:</p>
<pre><code class="language-python">@pytest.mark.parametrize("vpcs, flow_logs_responses, expected_statuses", [
    (
        [{"VpcId": "vpc-1"}], [{"FlowLogs": [{"FlowLogId": "fl-1"}]}],
        [CheckStatus.PASSED]
    ),
    (
        [{"VpcId": "vpc-1"}], [{"FlowLogs": []}],
        [CheckStatus.FAILED]
    ),
])
def test_vpc_flowlogs_varied_cases(self, vpcs, flow_logs_responses, expected_statuses):
    ...
</code></pre>
<hr>
<h3>🧼 <strong>Minor Suggestions</strong></h3>
<ul>
<li>
<p>The class docstring is good but can be more specific:<br>
<code inline="">"Test cases for checking whether Flow Logs are enabled for all VPCs."</code></p>
</li>
<li>
<p>Group related assertions:</p>
<pre><code class="language-python">assert len(report.resource_ids_status) == 2
assert {r.status for r in report.resource_ids_status} == {CheckStatus.FAILED}
</code></pre>
</li>
</ul>
<hr>
